vcpkg/maintainers/functions/vcpkg_make_configure.md

```diff
@@ -0,0 +1,166 @@
+
+### DISABLE_MSVC_WRAPPERS
+
+Disables the use of wrapper scripts for MSVC tools (cl.exe, windres).
+These wrappers translate and reorder command-line flags to accommodate the expected syntax and operational requirements of these tools. When this option is enabled, the `cl.exe` and `windres` tools will directly interpret command-line flags without any modifications.
+
+### NO_CPPFLAGS
+
+Disables the extraction of C preprocessor flags (`-D` and `-isystem`) from `CFLAGS` into `CPPFLAGS`.
+
+By default (when `NO_CPPFLAGS` is not set or is false), vcpkg automatically identifies and extracts these flags from the `CFLAGS` and `CXXFLAGS` variables, which typically contain general compiler options. It then consolidates these specific flags into a separate `CPPFLAGS` variable, dedicated to preprocessor and include path settings.
+
+However, if you explicitly set `NO_CPPFLAGS` to true, you override this default behavior. Vcpkg will then leave the `-D` and `-isystem` flags untouched in their original variables (`CFLAGS` and `CXXFLAGS`). 
+
+If set, the `CPPFLAGS` environment variable, typically used for specifying C preprocessor flags, will not be automatically populated based on vcpkg's settings. This can be useful for projects that do not use standard environment variables or have specific preprocessor requirements.
+
+### ADD_BIN_TO_PATH
+
+Adds the configure dependent `(debug/)bin` directory to the system path. This is useful if configure builds and runs executables with vcpkg integrated dependencies.
+
+### NO_DEFAULT_OPTIONS
+
+When set, this flag disables the automatic inclusion of the following default configure options provided by `vcpkg_make_configure`. Use this flag if you need full control over the configure command line:
+
+General Defaults:
+* --disable-silent-rules - Makes the build process output more verbose.
+* --verbose - Enables verbose output from the configure script.
+
+Library Linkage:
+* For dynamic library builds: --enable-shared and --disable-static
+* For static library builds: --disable-shared and --enable-static
+
+Installation Directories:
+* Adjusts standard directories (bin, lib, share, etc.) to conform with vcpkg's layout, ensuring correct installation paths.
+
+Platform-Specific Tweaks:
+* On Windows: Ensures compatibility with Windows-specific filesystem and library behaviors.
+* On Unix-like systems: Assures the script accommodates the typical filesystem layout and toolchain behaviors.
+
+Removing these defaults gives you the responsibility to specify all necessary configuration options manually, ensuring the project is correctly configured for building with vcpkg.
+
+### NO_MSVC_FLAG_ESCAPING
+
+By default, escape characters (e.g., -Xcompiler, -Xlinker) are added before compiler and linker flags when using MSVC. This is done to ensure that these flags, which might contain spaces or other special characters, are correctly interpreted by the underlying MSVC tools (cl.exe, link.exe)
```

Comment:
MSVC doesn't know `-Xcompiler` or `-Xlinker`, is there some other tool that reads these?

Reply:
No, this is for the wrappers. I'll explicitly mention that